ui-lib/src/main/java/co/electriccoin/zcash/ui/MainActivity.kt

```diff
@@ -85,6 +124,31 @@ class MainActivity : FragmentActivity() {
         setupUiContent()
 
         monitorForBackgroundSync()
+
+        lifecycleScope.launch {
+            addressBookRepository.googleSignInRequest.collect {
+                requestGoogleSignIn()
+            }
+        }
+
+        lifecycleScope.launch {
+            addressBookRepository.googleRemoteConsentRequest.collect { intent ->
+                googleConsentLauncher.launch(intent)
+            }
+        }
+    }
+
+    private fun requestGoogleSignIn() {
+        val googleSignInClient =
+            GoogleSignIn.getClient(
+                this@MainActivity,
+                GoogleSignInOptions
+                    .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                    .requestScopes(Scope(Scopes.DRIVE_APPFOLDER))
```

Comment:
`DRIVE_APPFOLDER` tells the API to use the hidden folder under the user's drive space, right?

Reply:
it's a request to access the user drive space :)

Reply:
Yes, I see. The `appDataFolder` is an app-specific folder and thus works as a hidden folder for regular user Drive access. You're handling it in `RemoteAddressBookDataSource`